starlette/applications.py

```diff
@@ -124,10 +132,15 @@ def mount(self, path: str, app: ASGIApp, name: str | None = None) -> None:
     def host(self, host: str, app: ASGIApp, name: str | None = None) -> None:
         self.router.host(host, app=app, name=name)  # pragma: no cover
 
-    def add_middleware(self, middleware_class: type, **options: typing.Any) -> None:
+    def add_middleware(
+        self,
+        middleware_class: typing.Callable[Concatenate[ASGIApp, P], typing.Any],
```

Comment:
```suggestion
        middleware_class: typing.Callable[Concatenate[ASGIApp, P], None],
```

Reply:
I'm afraid this suggestion is incorrect. 

Consider the following minimal example
```python
from typing import Callable


class Foo: pass

foo: Callable[..., None] = Foo
```

This will raise the following error in **mypy**:
```shell
 error: Incompatible types in assignment (expression has type "Type[Foo]", variable has type "Callable[..., None]")  [assignment]
```

`Class`es can be considered `Callable`s since they returns instances when called.